src/responder/common/negcache.c

```diff
@@ -861,6 +861,60 @@ int sss_ncache_check_locate_uid(struct sss_nc_ctx *ctx,
     return ret;
 }
 
+static char *locate_sid_str(struct sss_nc_ctx *ctx,
+                            struct sss_domain_info *dom,
+                            const char *sid)
+{
+    return talloc_asprintf(ctx,
+                           "%s/%s/%s/%s",
+                           NC_DOMAIN_ACCT_LOCATE_PREFIX,
+                           NC_SID_PREFIX,
+                           dom->name,
+                           sid);
+}
+
+int sss_ncache_check_locate_sid(struct sss_nc_ctx *ctx,
+                                struct sss_domain_info *dom,
+                                const char *sid)
+{
+    char *str;
+    int ret;
+
+    if (dom == NULL) {
+        return EINVAL;
+    }
+
+    str = locate_sid_str(ctx, dom, sid);
+    if (str == NULL) {
+        return ENOMEM;
+    }
+
+    ret = sss_ncache_set_str(ctx, str, false, false);
```

Comment:
Shouldn't this be a call to `sss_ncache_check_str()`?

Reply:
You are right, thanks.